### ISSUE TITLE
Inception behaves weirdly

### ISSUE THREAD

**Issue by nyarly** [samsalisbury/psyringe#5]
This is maybe a weird use case, but I can't find a less weird approach:

```
p := psyringe.New()
p.Add(func() *MyStruct{ return newMyStruct() })
p.Add(p)

var one, two struct{ 
  *pyringe.Psyringe
  *MyStruct
}

p.Inject(&one)
one.Psyringe.Inject(&two)
```

At this point, I would (wrongly) expect that
one.MyStruct is exactly two.MyStruct
because I would expect one.Psyringe to be p,
and for p's MyStruct to be a singleton

However, for reasons I haven't dug into, this isn't the case: the two MyStructs are similar, but each its own allocation.

**Comment by nyarly** [comment]
Further examination determines that this is a cloning problem. PR follows.

**Comment by samsalisbury** [comment]
I believe this is resolved by 454e8c1d6064170696ec3aeb4bab7c6ea839d495.
